mitsurunner.yaml

```diff
@@ -81,96 +90,172 @@ sensor:
+    - switch.turn_on: gpio_relay
+
     - lambda: !lambda |-
-        id(mqtt_client).publish(TOPIC_DEFROST_HACK_STATE, to_string(0.0));
-    - delay: 30min    
+        id(G_state) = ST_IDLE;
+        id(G_state_time_passed) = false;
+
+
+  - id: EnterTempExceeded
+    mode: single
+    then:
+    #Stop other state scripts
+    - script.stop: EnterIdle
+    - script.stop: EnterStartDefrosting
+    - script.stop: EnterDefrostingStarted
+    - script.stop: EnterHeatingMinTime
     - switch.turn_on: gpio_relay
+
     - lambda: !lambda |-
-        id(mqtt_client).publish(TOPIC_DEFROST_HACK_STATE, to_string(1.0));
+        id(G_state) = ST_TEMP_EXCEEDED;
+        id(G_state_time_passed) = false;
+
+    - delay: 5min
+    - lambda: !lambda |-
+        id(G_state_time_passed) = true;
 
 
+  - id: EnterStartDefrosting
+    mode: single
+    then:
+    #Stop other state scripts
+    - script.stop: EnterIdle
+    - script.stop: EnterTempExceeded
+    - script.stop: EnterDefrostingStarted
+    - script.stop: EnterHeatingMinTime
+    - switch.turn_off: gpio_relay
 
-# scipt implementing timer for minimum heating time
-# G_min_heating_time_passed is used through this timer to prevent defrosting too quickly after previous defrosting
-  - id: start_min_heating_time_timer
-    mode: single #will be started only if there were no ongoing defrosting already
-    then:  
     - lambda: !lambda |-
-        id(G_min_heating_time_passed) = false;
-        id(mqtt_client).publish(TOPIC_IS_HEATING_MIN_TIME_PASSED, to_string(0.0));
-    - delay: 50min 
+        id(G_state) = ST_START_DEFROSTING;
+        id(G_state_time_passed) = false;
+
+    - delay: 10min
     - lambda: !lambda |-
-        id(G_min_heating_time_passed) = true; 
-        id(mqtt_client).publish(TOPIC_IS_HEATING_MIN_TIME_PASSED, to_string(1.0));
-        
-# script implementing timer preventing defrosting if temperature delta excess defrosting treshold limit just short time. (filters out peaks)
-  - id: start_excess_min_time_timer
-    mode: single #will be started only if there were no ongoing defrosting already
-    then:  
+        id(G_state_time_passed) = true;
+
+
+  - id: EnterDefrostingStarted
+    mode: single
+    then:
+    #Stop other state scripts
+    - script.stop: EnterIdle
+    - script.stop: EnterTempExceeded
+    - script.stop: EnterStartDefrosting
+    - script.stop: EnterHeatingMinTime
+    - switch.turn_off: gpio_relay
+
+    - script.stop: schedule_forced_defrosting
+    - script.execute: schedule_forced_defrosting
+
     - lambda: !lambda |-
-        id(G_excess_min_time_passed) = false;
-        id(mqtt_client).publish(TOPIC_IS_EXCESS_MIN_TIME_PASSED, to_string(0.5));
-    - delay: 5min 
+        id(G_state) = ST_DEFROSTING_STARTED;
+        id(G_state_time_passed) = false;
+
+    - delay: 30min
     - lambda: !lambda |-
-        id(G_excess_min_time_passed) = true; 
-        id(mqtt_client).publish(TOPIC_IS_EXCESS_MIN_TIME_PASSED, to_string(1.0));
```

Comment:
Needed for testing/verifying fuctinoality

Reply:
These values can be seen from the state mqtt message but It's better to keep them for backwards compatibility, so I'll restore them.